site/src/hooks/usePreferredProxy.ts

```diff
@@ -0,0 +1,20 @@
+import { Region } from "api/typesGenerated"
+import { useDashboard } from "components/Dashboard/DashboardProvider"
+
+export const usePreferredProxy = (): Region | undefined => {
+  const dashboard = useDashboard()
+  // Only use preferred proxy if the user has the moons experiment enabled
+  if (!dashboard?.experiments.includes("moons")) {
+    return undefined
+  }
+
+  const str = localStorage.getItem("preferred-proxy")
```

Comment:
This localStorage should only store the ID, and it should get the regions list on startup (lazily perhaps)

Storing the whole JSON object means if any changes are made serverside they won't get considered on clients

Reply:
Yea, I am keeping the proxy for now for naming reasons.
The new context will fetch all the regions and update the proxy values if they are stale.